nexus/reconfigurator/planning/src/blueprint_editor/sled_editor.rs

```diff
@@ -686,6 +721,84 @@ impl ActiveSledEditor {
         Ok(())
     }
 
+    /// Update a sled's mupdate override field based on the mupdate override
+    /// provided by inventory.
+    pub fn ensure_mupdate_override(
+        &mut self,
+        inv_mupdate_override_id: Option<MupdateOverrideUuid>,
+    ) -> Result<EnsureMupdateOverrideAction, SledEditError> {
+        match (inv_mupdate_override_id, *self.remove_mupdate_override.value()) {
+            (Some(inv_override), Some(bp_override))
+                if inv_override == bp_override =>
+            {
+                // If the inventory and blueprint overrides are the same, the
+                // sled agent hasn't yet removed the override. Nothing to do at
+                // the moment.
+                Ok(EnsureMupdateOverrideAction::NoAction {
+                    mupdate_override: Some(inv_override),
+                })
+            }
+            (Some(inv_override), bp_override) => {
+                // Inventory says there's an override in place, but the
+                // blueprint doesn't (or has a different override in place).
+                // This means that a MUPdate happened since we last did
+                // blueprint planning.
+                //
+                // Set the blueprint's remove_mupdate_override.
+                self.set_remove_mupdate_override(Some(inv_override));
+                // Set all zone image sources to InstallDataset. This is an
+                // acknowledgement of the current state of the world.
+                let zone_ids: Vec<_> = self
+                    .zones(BlueprintZoneDisposition::is_in_service)
+                    .map(|zone| (zone.id, zone.kind()))
+                    .collect();
+
+                let mut zones = IdOrdMap::with_capacity(zone_ids.len());
+                for (zone_id, kind) in zone_ids {
+                    let old_image_source = self.zones.set_zone_image_source(
+                        &zone_id,
+                        BlueprintZoneImageSource::InstallDataset,
+                    )?;
```

Comment:
[RFD 556](https://rfd.shared.oxide.computer/rfd/0556) says:

> Wherever the planner uses the target release, it is instead ignored if its generation number is not greater than min_release_generation (if set).

As discussed in Tuesday's watercooler it's a bit more complex than that -- what we want to do is to only use the install dataset on sleds that have been mupdated.

I think I want to make the claim that this code may actually be sufficient as it stands. I don't think we need to try and do any other redirects other than this one (which is admittedly edge-triggered), as long as we prevent new zones from being set up at all while the system is recovering from the mupdate.